### PR TITLE
chore: update status-go for StoreNodeRequestManager

### PR DESCRIPTION
**Pull request by igor-sirotin**
### Summary

PR to test https://github.com/status-im/status-go/pull/4446

### Testing notes

I refactored the fetching communities part. The API is kept the same.

status: ready

### CODE REVIEW

**Comment by churik** [comment]
@igor-sirotin 
Tested with build from https://github.com/status-im/status-desktop/pull/13012

### ISSUE 1: The link with the token-gated community does not detect permissions on mobile

**Steps:**
1) create token-gated community on desktop (in my case 10 SNT to become a member, [link](https://status.app/c/CxeACi0KC2ZmZmZmZiBnZ2dnEg5qbGtubGpibmxqbmpsbhgBIgcjNDM2MERGKgMxKiYD#zQ3shXp8mJbfHkJQ4hqSehYSErucMxQVGcS7r4yw1Akb6nKzn)
2) send a link to the community to mobile 
3) open it on mobile


| **Expected result:** (current nightly,13.12.23) | **Actual result:** |
|-|-|
| IMAGE 2023-12-13 16:58:07 | MAGE 2023-12-13 16:58:40 |


**OS:** IOS, Android

### ISSUE 2 (desktop):  preview is not generated when paste it as another user

**Steps:**
1) User1: create community on desktop ([link](https://status.app/c/ixGACiIKB2hlbGxvX3kSB2hlbGxvX3kYAyIHIzQzNjBERioDMCktAw==#zQ3shX3HakX9bGLzZZMi82ucQF6RRGacjdCTs7qJ4WKEdNcpg)
2) User2: paste this link to any chat

**Expected result:** 
community info will be fetched

**Actual result:**
no community info, nothing happens when tapping on the link

https://github.com/status-im/status-mobile/assets/4557972/01b7b1d1-b693-4cd2-b50e-f28dcbb7e8c2
```
INF 2023-12-13 17:05:00.675+01:00 mailserver available                       topics="mailservers-service" tid=229782 file=service.nim:156
INF 2023-12-13 17:05:00.675+01:00 history request started                    topics="mailservers-service" tid=229782 file=service.nim:166 numBatches=1
Warning: libpng warning: iCCP: known incorrect sRGB profile (:0, )
Warning: libpng warning: iCCP: known incorrect sRGB profile (:0, )
Warning: libpng warning: iCCP: known incorrect sRGB profile (:0, )
Warning: libpng warning: iCCP: known incorrect sRGB profile (:0, )
INF 2023-12-13 17:05:01.117+01:00 history request completed                  topics="mailservers-service" tid=229782 file=service.nim:171
2023-12-13T17:05:20+0100 unset [github.com/anacrolix/torrent.(*Client).NewAnacrolixDhtServer.func2:406]: dht server on 0.0.0.0:55504 (node id b302b35612ba417d42c8a6f5a56150a3695c0268) completed bootstrap ({NumAddrsTried:52 NumResponses:21})
2023-12-13T17:05:23+0100 unset [github.com/anacrolix/torrent.(*Client).NewAnacrolixDhtServer.func2:406]: dht server on [::]:55504 (node id d80ba1d4bf7b841aa520c7cc912c0d8b7155a5e6) completed bootstrap ({NumAddrsTried:59 NumResponses:26})
INF 2023-12-13 17:06:08.387+01:00 history request started                    topics="mailservers-service" tid=229782 file=service.nim:166 numBatches=1
INF 2023-12-13 17:06:08.900+01:00 history request completed                  topics="mailservers-service" tid=229782 file=service.nim:171

```

**Comment by churik** [comment]
### ISSUE 3 (desktop): can't import the community on desktop 

1) Import community > Paste ` zQ3shX3HakX9bGLzZZMi82ucQF6RRGacjdCTs7qJ4WKEdNcpg`


**Expected result:**
community is fetched

**Actual result:**
Can't find the community 

**Log:** [import_comm.log](https://github.com/status-im/status-mobile/files/13663274/import_comm.log)

[geth.log](https://github.com/status-im/status-mobile/files/13663276/geth.log)

image


**OS:** MacOSX

**Comment by igor-sirotin** [comment]
> ISSUE 2 (desktop): preview is not generated when paste it as another user
> ISSUE 3 (desktop): can't import the community on desktop

@churik @anastasiyaig 

TLDR: the problem here is that your account is using `shards.test` fleet and the community is known on `status.prod` fleet.

1. Yes, `shards.test` should be the new default fleet.
2. It seems that right now `status.prod` is the default shard in status-desktop.
3. It's obvious from the logs that your app is using `shards.test` fleet
```go
INF 2023-12-13 17:29:23.446+01:00 initMailservers                            topics="mailservers-service" tid=252348 file=service.nim:190 name=store-01.do-ams3.shards.test nodeAddress=/dns4/store-01.do-ams3.shards.test.statusim.net/tcp/30303/p2p/16Uiu2HAmAUdrQ3uwzuE4Gy4D56hX6uLKEeerJAnhKEHZ3DxF1EfT
INF 2023-12-13 17:29:23.447+01:00 initMailservers                            topics="mailservers-service" tid=252348 file=service.nim:190 name=store-02.gc-us-central1-a.shards.test nodeAddress=/dns4/store-02.gc-us-central1-a.shards.test.statusim.net/tcp/30303/p2p/16Uiu2HAmJnVR7ZzFaYvciPVafUXuYGLHPzSUigqAmeNw9nJUVGeM
INF 2023-12-13 17:29:23.447+01:00 initMailservers                            topics="mailservers-service" tid=252348 file=service.nim:190 name=store-01.ac-cn-hongkong-c.shards.test nodeAddress=/dns4/store-01.ac-cn-hongkong-c.shards.test.statusim.net/tcp/30303/p2p/16Uiu2HAm2M7xs7cLPc3jamawkEqbr7cUJX11uvY7LxQ6WFUdUKUT
INF 2023-12-13 17:29:23.447+01:00 initMailservers                            topics="mailservers-service" tid=252348 file=service.nim:190 name=store-02.ac-cn-hongkong-c.shards.test nodeAddress=/dns4/store-02.ac-cn-hongkong-c.shards.test.statusim.net/tcp/30303/p2p/16Uiu2HAm9CQhsuwPR54q27kNj9iaQVfyRzTGKrhFmr94oD8ujU6P
INF 2023-12-13 17:29:23.447+01:00 initMailservers                            topics="mailservers-service" tid=252348 file=service.nim:190 name=store-02.do-ams3.shards.test nodeAddress=/dns4/store-02.do-ams3.shards.test.statusim.net/tcp/30303/p2p/16Uiu2HAm9aDJPkhGxc2SFcEACTFdZ91Q5TJjp76qZEhq9iF59x7R
INF 2023-12-13 17:29:23.447+01:00 initMailservers                            topics="mailservers-service" tid=252348 file=service.nim:190 name=store-01.gc-us-central1-a.shards.test nodeAddress=/dns4/store-01.gc-us-central1-a.shards.test.statusim.net/tcp/30303/p2p/16Uiu2HAmMELCo218hncCtTvC2Dwbej3rbyHQcR8erXNnKGei7WPZ
```
4. Probably this happened because you have not a new account and it was created with `shards.test` fleet.

Since I tested this with a brand new account, I have `status.prod` fleet selected. Hence I'm successfully fetching the community in both cases and can't reproduce these 2 bugs.

Screenshot 2023-12-13 at 20 46 54

Screenshot 2023-12-13 at 20 47 12

**Comment by churik** [comment]
@igor-sirotin 
thank you!
I'll re-test issue 2-3,the issue 1 is valid for sure, thanks!

**Comment by igor-sirotin** [comment]
> @igor-sirotin 
> 
> thank you!
> 
> I'll re-test issue 2-3,the issue 1 is valid for sure, thanks!

Working on this, having issues building status-mobile 😄
With status-desktop can't reproduce.

**Comment by churik** [comment]
Ok, so ISSUE 1 is about the way we are getting the updates for communities.
It means that if we open this link once and permissions/description etc changed afterwards, there is no way for this user to get these updates - the link will stay the same and show outdated info.

**Comment by churik** [comment]
@igor-sirotin no issues from my POV anymore, PR is ready to merge

**Comment by igor-sirotin** [comment]
Closing as status-go was already updated to a newer version.
